### PR TITLE
JDK-8260707: java/lang/instrument/PremainClass/InheritAgent0100.java times out

### PR DESCRIPTION
**Pull request by tstuefe**
Hi,

may I please have reviews for this trivial fix.

We see timeouts with this test on slow large (memory wise) AIX machines, as well as large core files.

This test is a negative test which tests that the VM corectly recognizes an error condition and aborts. Code goes through jni_FatalError()->os::abort(), writes a core and aborts the process.

No parameters are given for that VM invocation, so heap size depends on machine size. On my Linux box, the generated core takes about 500m. On AIX we see cores of 16G and more.

The difference between AIX and other platforms is that AIX does not have the notion of "committing" memory (we have no MAP_NORESERVE flag), so all mmapped memory counts toward the commit charge from the moment it is mapped. That makes core files on AIX annoyingly large. I'd expect a similar behavior on other platforms with overcommit disabled.

The fix is to run the test without CreateCoredumpOnCrash. For good measure, I also reduced the heap size to 128m.

Thanks to @ArnoZeller for figuring that one out.

/contributor add @ArnoZeller

-----
Tests: manual jtreg tests, verifying that no cores are written.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8260707](https://bugs.openjdk.java.net/browse/JDK-8260707): java/lang/instrument/PremainClass/InheritAgent0100.java times out

### Reviewers
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**) ⚠️ Review applies to 746972071bd7fab1359e670c0c60be54268250e9
 * [Serguei Spitsyn](https://openjdk.java.net/census#sspitsyn) (@sspitsyn - **Reviewer**)
 * [Daniel D. Daugherty](https://openjdk.java.net/census#dcubed) (@dcubed-ojdk - **Reviewer**)

### Contributors
 * Arno Zeller `<azeller@openjdk.org>`

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2332/head:pull/2332`
`$ git checkout pull/2332`

### CODE REVIEW

**Review by dholmes-ora — APPROVED** [review]
Hi Thomas,

This seems fine. Some of the tests that use NegativeAgentRunner already add -XX:-CreateCoreDumpOnCrash (somewhat pointlessly I think) on their @run lines - so these could be removed. (They could all be converted to driver mode as well AFAICS but that may be asking a bit much for this change. :))
Thanks,
David

**Review by sspitsyn — APPROVED** [review]
Hi Thomas,
The fix looks good. Thank you fixing it.
I've submitted mach5 test job with 40 runs of the java/lang/instrument/PremainClass tests.
All are passed while they were failing before.
Also, David is right, the option -XX:-CreateCoreDumpOnCrash can be removed from tests:
 test/jdk/java/lang/instrument/PremainClass/NoPremainAgent.java
 test/jdk/java/lang/instrument/PremainClass/ZeroArgPremainAgent.java
 
 But I can file a separate bug on it.
 Also, I've closed bug https://bugs.openjdk.java.net/browse/JDK-8260469 as a dup of this one.
  
Thanks,
Serguei

**Comment by tstuefe** [comment]
David, Sergey,

thanks for the reviews! I removed the superfluous options and re-ran the affected tests locally, all seemed fine. No cores popped up either. I will wait the 24hrs, then integrate.

Cheers, Thomas

File: test/jdk/java/lang/instrument/NegativeAgentRunner.java [hunk 1]
```diff
@@ -36,7 +36,7 @@ public static void main(String argv[]) throws Exception {
         String agentClassName = argv[0];
         String excepClassName = argv[1];
         ProcessBuilder pb = ProcessTools.createJavaProcessBuilder(
-                "-javaagent:" + agentClassName + ".jar",
+                "-javaagent:" + agentClassName + ".jar", "-Xmx128m", "-XX:-CreateCoredumpOnCrash",
```
Comment (dcubed-ojdk) [inline test/jdk/java/lang/instrument/NegativeAgentRunner.java:39]: Thanks for also reducing the stack size.

**Review by dcubed-ojdk — APPROVED** [review]
Thumbs up! Thanks for taking care of this.

**Comment by dcubed-ojdk** [comment]
I think this is a trivial fix and does not need to wait 24 hours.

**Comment by tstuefe** [comment]
> I think this is a trivial fix and does not need to wait 24 hours.

Thanks!

> The update looks good.

Thanks!

/integrate

**Pull request closed without merging**